sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2527,6 +2527,17 @@ object functions {
     Overlay(src.expr, lit(replaceString).expr, lit(pos).expr, lit(len).expr)
   }
 
+  /**
+   * Overlay the specified portion of `src` with `replaceArray`,
+   *  starting from byte position `pos` of input byte array and proceeding for `len` bytes.
+   *
+   * @group string_funcs
+   * @since 3.0.0
+   */
+  def overlay(src: Column, replaceArray: Array[Byte], pos: Int, len: Int): Column = withExpr {
+    Overlay(src.expr, lit(replaceArray).expr, lit(pos).expr, lit(len).expr)
```

Comment:
1. let's add it into Python and R side too. This can be done separately.
2. signature should be `overlay(src: Column, replaceArray: Column, pos: Column, len: Column)`. See the comments above in `functions.scala` file